src/CommunityToolkit.Maui/Views/RatingView/RatingView.Shared.cs

```diff
@@ -0,0 +1,380 @@
+				image.StrokeThickness = StrokeThickness;
+
+				image.Aspect = Stretch.Uniform;
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+			}
+
+
+			if (ShouldAllowRating)
+			{
+				var tapGestureRecognizer = new TapGestureRecognizer();
+				tapGestureRecognizer.Tapped += OnShapeTapped;
+				image.GestureRecognizers.Add(tapGestureRecognizer);
+			}
+
+			innerContent?.Children.Add(image);
+			innerContent?.SetColumn(image, i);
+
+			shapes[i] = image;
+		}
+
+
+		UpdateDraw();
+	}
+
+	void ReDraw()
+	{
+		innerContent?.Children.Clear();
+
+		innerContent?.ColumnDefinitions.Clear();
+
+		shapes = new Microsoft.Maui.Controls.Shapes.Path[MaximumRating];
+
+		for (int i = 0; i < MaximumRating; i++)
+		{
+			innerContent?.ColumnDefinitions.Add(new ColumnDefinition { Width = Size });
+
+			Microsoft.Maui.Controls.Shapes.Path image = new();
+#pragma warning disable CS8600 // Converting null literal or possible null value to non-nullable type.
+			image.Data = (Geometry)new PathGeometryConverter().ConvertFromInvariantString(shape);
+#pragma warning restore CS8600 // Converting null literal or possible null value to non-nullable type.
+			if (i <= CurrentRating)
+			{
+
+				image.Fill = FilledBackgroundColor;
+				image.Stroke = FilledBackgroundColor;
+				image.Aspect = Stretch.Uniform;
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+			}
+			else
+			{
+				image.Fill = EmptyBackgroundColor;
+				image.Stroke = StrokeColor;
+
+				image.StrokeLineJoin = PenLineJoin.Round;
+				image.StrokeLineCap = PenLineCap.Round;
+				image.StrokeThickness = StrokeThickness;
+
+				image.Aspect = Stretch.Uniform;
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+			}
+
+
+			if (ShouldAllowRating)
+			{
+				var tapGestureRecognizer = new TapGestureRecognizer();
+				tapGestureRecognizer.Tapped += OnShapeTapped;
+				image.GestureRecognizers.Add(tapGestureRecognizer);
+			}
+
+			innerContent?.Children.Add(image);
+			innerContent?.SetColumn(image, i);
+
+			shapes[i] = image;
+		}
+
+		UpdateDraw();
+	}
+
+	void UpdateDraw()
+	{
+		for (int i = 0; i < MaximumRating; i++)
+		{
+			Microsoft.Maui.Controls.Shapes.Path image = shapes[i];
+
+			if (CurrentRating >= i + 1)
+			{
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+				image.StrokeLineJoin = PenLineJoin.Round;
+				image.StrokeThickness = StrokeThickness;
+				image.Stroke = FilledBackgroundColor;
+			}
+			else
+			{
+				image.HeightRequest = Size;
+				image.WidthRequest = Size;
+				if (CurrentRating % 1 == 0)
+				{
+					image.Fill = EmptyBackgroundColor;
+					image.Stroke = StrokeColor;
+					image.StrokeThickness = StrokeThickness;
+					image.StrokeLineJoin = PenLineJoin.Round;
+					image.StrokeLineCap = PenLineCap.Round;
+				}
+				else
+				{
+					var fraction = CurrentRating - Math.Floor(CurrentRating);
+					var element = shapes[(int)(CurrentRating - fraction)];
+					if (element != null)
```

Comment:
we use pattern matching to check for `null`
```suggestion
					if (element is not null)
```

Reply:
Thank you for the tip